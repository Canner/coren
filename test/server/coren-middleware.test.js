const corenMiddleware = require('../../server/coren-middleware');

describe('coren-middleware', () => {
  it('init test', () => {
    const myMock = jest.fn();
    const middleware = corenMiddleware('./test/server/testExample/corenMiddleWareTest');
    middleware({}, {}, myMock);
    expect(myMock).toBeCalled();
  });

  describe('test res function', () => {
    let middleware;
    let res;
    let req;
    let mockSend;
    beforeEach(() => {
      mockSend = jest.fn();
      req = {};
      res = {send: mockSend};
      middleware = corenMiddleware('./test/server/testExample/corenMiddleWareTest');
      middleware(req, res, jest.fn());
    });

    describe('sendCoren', () => {
      it('/', () => {
        res.sendCoren('/');
        expect(mockSend.mock.calls[0][0]).toMatchSnapshot();
      });
      it('/users', () => {
        res.sendCoren('/users');
        expect(mockSend.mock.calls[0][0]).toMatchSnapshot();
      });
      it('/users/1', () => {
        res.sendCoren('/users/1');
        expect(mockSend.mock.calls[0][0]).toMatchSnapshot();
      });
    });

    it('setHead', () => {
      res.setHead(function($head) {
        $head.append('<script>hihihihi</script>');
      });
      res.sendCoren('/');
      expect(mockSend.mock.calls[0][0]).toMatchSnapshot();
    });

    it('setPreloadedState', () => {
      res.setPreloadedState({auth: false, user: 'john'});
      res.sendCoren('/');
      expect(mockSend.mock.calls[0][0]).toMatchSnapshot();
    });

    it('setHead & setPreloadedState', () => {
      res.setHead(function($head) {
        $head.append('<script>hihihihi</script>');
      });
      res.setPreloadedState({auth: false, user: 'john'});
      res.sendCoren('/');
      expect(mockSend.mock.calls[0][0]).toMatchSnapshot();
    });
  });
});
