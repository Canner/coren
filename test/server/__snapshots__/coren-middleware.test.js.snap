// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coren-middleware test res function sendCoren index 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <title>home</title>
  <script data-coren=\\"\\">
    window.__PRELOADED_STATE__ = {\\"auth\\":false}
  </script>
  <meta name=\\"description\\" content=\\"home description\\">
</head>

<body>
  <div id=\\"root\\">
    <button class=\\"hi\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-952690291\\">index file!</button>
  </div>



</body></html>"
`;

exports[`coren-middleware test res function sendCoren index/users 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <title>home</title>
  <script data-coren=\\"\\">
    window.__PRELOADED_STATE__ = {\\"auth\\":false}
  </script>
  <meta name=\\"description\\" content=\\"home description\\">
</head>

<body>
  <div id=\\"root\\">
    <button class=\\"hi\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-952690291\\">users file!</button>
  </div>



</body></html>"
`;

exports[`coren-middleware test res function sendCoren index/users/1 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <title>home</title>
  <script data-coren=\\"\\">
    window.__PRELOADED_STATE__ = {\\"auth\\":false}
  </script>
  <meta name=\\"description\\" content=\\"home description\\">
</head>

<body>
  <div id=\\"root\\">
    <button class=\\"hi\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-952690291\\">users file! #1</button>
  </div>



</body></html>"
`;

exports[`coren-middleware test res function setHead & setPreloadedState 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <title>home</title>
  <script data-coren=\\"\\">window.__PRELOADED_STATE__ = {\\"auth\\":false,\\"user\\":\\"john\\"}</script>
  <meta name=\\"description\\" content=\\"home description\\">
<script>hihihihi</script></head>

<body>
  <div id=\\"root\\">
    <button class=\\"hi\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-952690291\\">index file!</button>
  </div>



</body></html>"
`;

exports[`coren-middleware test res function setHead 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <title>home</title>
  <script data-coren=\\"\\">
    window.__PRELOADED_STATE__ = {\\"auth\\":false}
  </script>
  <meta name=\\"description\\" content=\\"home description\\">
<script>hihihihi</script></head>

<body>
  <div id=\\"root\\">
    <button class=\\"hi\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-952690291\\">index file!</button>
  </div>



</body></html>"
`;

exports[`coren-middleware test res function setPreloadedState 1`] = `
"<!DOCTYPE html><html><head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <title>home</title>
  <script data-coren=\\"\\">window.__PRELOADED_STATE__ = {\\"auth\\":false,\\"user\\":\\"john\\"}</script>
  <meta name=\\"description\\" content=\\"home description\\">
</head>

<body>
  <div id=\\"root\\">
    <button class=\\"hi\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-952690291\\">index file!</button>
  </div>



</body></html>"
`;
